### ISSUE TITLE
Feature Idea: Subtasks

### ISSUE THREAD

**Issue by ZevSwe** [mattermost/focalboard#252]
**Summary:**
I know markdown supports checklists but I would want to be able to sort tasks in a better way. Say we have a project that consists of stages planning, building, testing, reporting, lessons learned and that planning is made up of brainstorming, selecting ideas, scheduling and reporting is built of writing draft, filling in missing information, fixing appearance, fixing references, proof reading, editing, publishing. How would I sort this in a way so that not everything is just added in a long list of tasks? Maybe by adding a property and sort by it? When working in larger projects this would be very useful.

**How important this is to me and why:**

Importance: Medium
This would make the product much more useful, but it might be hard to implement in a good way. One can find work arounds but a native solution would be very nice.

Use cases:
1. Divide large projects into smaller parts to get a better understanding of progress.

**Comment by codingthat** [comment]
A template task that can contain subtasks like this would be really powerful.  Arbitrary nesting (while verifying there are no loops) would be a very flexible way to accomplish this.

**Comment by chenilim** [comment]
Thanks @ZevSwe! Now that the checkbox feature is in [v0.6.5](https://github.com/mattermost/focalboard/discussions/318), would you consider this resolved? Or are there additional capabilities around subtasks? e.g. assigning subtasks to other people?

**Comment by ZevSwe** [comment]
Hmm good question. When it comes to checklists I've sent an idea in #288. When it comes to tasks and subtasks and even subtasks assigned to other people one could ask if the subtasks should not be separate cards. If so, how does one link them together and how does one show what is part of what master-ticket? This is very much a usecase for any team that wants to use this as their kanban board.

**Comment by chenilim** [comment]
Good point about linking cards. I created feature idea #324 to track that. Feel free to add to it. At a basic level, this could be implemented with a new **Card link** content type.

**Comment by chmac** [comment]
The checkboxes feature is a great start. The thing that's obviously missing for me (inspired by GitHub and Trello) is to be able to see how many checkboxes are closed and how many are open. Here's what I mean from trello:

<img width="265" alt="Screenshot 2021-08-18 at 12 42 13" src="https://user-images.githubusercontent.com/690997/129884691-3723bdbd-865c-4e2e-a669-38ef0f7c21ae.png">

Having that 0/3 count, and being able to somehow filter on it, would make the checkboxes super useful I think.

**Comment by chmac** [comment]
It would be great if checkboxes inside markdown blocks were also considered "checkboxes". In some ways, this would potentially solve #288 (allowing hierarchical lists). It's a bit weird from a UX perspective that 2 of these are checkboxes but the first 3 are not clickable:

<img width="258" alt="Screenshot 2021-08-18 at 12 45 35" src="https://user-images.githubusercontent.com/690997/129885212-3f44cca5-d9a5-48f1-a0c6-655d58241fbf.png">

**Comment by sebseb538** [comment]
Hi, 
Is it possible to have an update on this beacause we really need subtasks in how boards.

**Comment by alexanderadam** [comment]
It's probably unlikely that someone will implement this [unless the community is _really_ taking over](https://github.com/mattermost/focalboard#warning-announcements):

> _Mattermost developers will not be adding any new enhancements or bug fixes beyond September 15th, 2023._

Regarding updates and new features [Vikunja](https://vikunja.io/) might be a better choice for now and [it already has subtasks](https://vikunja.io/docs/task-relation-kinds/).

**Comment by osfast** [comment]
+1 would love this feature

**Comment by imanmagomedov** [comment]
Any updates?

**Comment by chmac** [comment]
Focalboard is no longer maintained, I've switched to planka and it works great.
